docs/concepts.md

```diff
@@ -0,0 +1,56 @@
+# yb Concepts
+
+## Package
+
+A _package_ is a collection of related files and targets. A package is a
+directory that contains a .yourbase.yml file directly inside it. Files can only
+belong to a single package. A package may contain subpackages in subdirectories,
+but files within those subdirectories belong to the subpackage, not the parent
+package.
+
+## Target
+
+A _target_ is a buildable thing &mdash; it could be a service, a library,
+a binary, etc. A target has four _phases_ (all optional, but at least one must
+be present):
+
+-  Build
+-  Test
+-  Execute
+-  Release
+
+Each phase, if present, has a list of one or more commands to run. The build
+phase can declare output files. Output files will be accessible to other targets
+that directly depend on this target.
```

Comment:
(unrelated to Amy's above comments, just on the same line) `build` is explicitly allowed to output files, but `test` is not. What about code coverage output?